src/test/java/org/snomed/snowstorm/core/data/services/BranchMergeServiceTest.java

```diff
@@ -4245,6 +4245,338 @@ void testPromotionDoesNotResultInExtensionStealingAssociationMember() throws Ser
+				.addDescription(new Description("Vehicle").setTypeId(SYNONYM).setCaseSignificance(ci).setAcceptabilityMap(intPreferred))
+				.addAxiom(new Relationship(ISA, SNOMEDCT_ROOT))
+				.setModuleId(MODEL_MODULE);
+		concept = conceptService.create(concept, intMain);
+		String vehicleId = concept.getConceptId();
+
+		// 2. Version
+		codeSystem = codeSystemService.find("SNOMEDCT");
+		codeSystemService.createVersion(codeSystem, 20220131, "20220131");
+
+		// 3. Create Project
+		String projectA = "MAIN/projectA";
+		branchService.create(projectA);
+
+		// 4. Create Task A (for inactivating the Concept)
+		String taskA = "MAIN/projectA/taskA";
+		branchService.create(taskA);
+
+		// 5. Create Task B (for changing the Concept's module)
+		String taskB = "MAIN/projectA/taskB";
+		branchService.create(taskB);
+
+		// 6. On Task A, inactivate the Concept.
+		concept = conceptService.find(vehicleId, taskA);
+		concept.setActive(false);
+		concept.updateEffectiveTime();
+		concept.setInactivationIndicator("AMBIGUOUS");
+		conceptService.update(concept, taskA);
+
+		// 7. On Task B, change the Concept's module.
+		concept = conceptService.find(vehicleId, taskB);
+		concept.setModuleId(CORE_MODULE);
+		conceptService.update(concept, taskB);
+
+		// 8. Promote Task A to CodeSystem
+		branchMergeService.mergeBranchSync(taskA, projectA, Collections.emptySet());
+		branchMergeService.mergeBranchSync(projectA, intMain, Collections.emptySet());
+
+		// 9. Version
+		codeSystem = codeSystemService.find("SNOMEDCT");
+		codeSystemService.createVersion(codeSystem, 20220228, "20220228");
+
+		// 10. Rebase Project
+		review = getMergeReviewInCurrentState(intMain, projectA);
+		conflicts = reviewService.getMergeReviewConflictingConcepts(review.getId(), new ArrayList<>());
+		assertEquals(0, conflicts.size());
+		reviewService.applyMergeReview(review);
+
+		// 11. Rebase Task B
+		review = getMergeReviewInCurrentState(projectA, taskB);
+		conflicts = reviewService.getMergeReviewConflictingConcepts(review.getId(), new ArrayList<>());
+		assertEquals(1, conflicts.size());
+
+		for (MergeReviewConceptVersions conflict : conflicts) {
+			Concept autoMergedConcept = conflict.getAutoMergedConcept();
+
+			reviewService.persistManuallyMergedConcept(review, Long.parseLong(vehicleId), autoMergedConcept);
+		}
+
+		reviewService.applyMergeReview(review);
+
+		// 12. Assert
+		concept = conceptService.find(vehicleId, taskB);
+		assertEquals(20220228, concept.getReleasedEffectiveTime()); // Concept has moved along
+		assertNull(concept.getEffectiveTimeI()); // Changes have been applied to the Concept
+		assertEquals(CORE_MODULE, concept.getModuleId()); // This is what changed on LHS
+		assertTrue(concept.isActive()); // Concept change should result in re-activation
+	}
+
+	@Test
+	void testRebasingVersionedAxiomChange() throws ServiceException, InterruptedException {
+		String intMain = "MAIN";
```

Comment:
@CoderMChu this is another test that fails on develop but passes on this feature branch.

Reply:
Develop ends up with 2 axioms and lost release statuses.